src/Compiler/Microsoft.AspNetCore.Razor.Language/test/MetadataCollectionTests.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Language;
+
+public class MetadataCollectionTests
+{
+    [Theory]
+    [InlineData(0)]
+    [InlineData(1)]
+    [InlineData(2)]
+    [InlineData(3)]
+    [InlineData(4)]
+    [InlineData(8)]
+    [InlineData(16)]
+    [InlineData(32)]
+    [InlineData(64)]
+    public void CreateAndCompareCollections(int size)
```

Comment:
Some tests of other behaviors, like duplicate keys when creating from a list of kvp, would be appreciated.